packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -77,7 +78,11 @@ export interface SdkOperationGroup {
   service: Namespace;
 }
 
-interface SdkTypeBase {
+interface DecoratedType {
+  decorators: Record<string, Record<string, any>>;
```

Comment:
I get that as decorators are open-ended it's difficult to have strongly typed models. However, this is very open-ended and without any docs/examples, callers are left to examining the contents in the debugger to understand how the data is structured.

E.g. in one of the XML tests I see this.

`TypeSpec.Xml.@attribute`

How is that value formulated, what's the pattern? Can I tell from looking at the docs for that decorator? Is the contract formalized and stable?

Reply:
yes. the open-ended list is hard to use. but the main purpose for this feature is we want to prevent adding properties to tcgc types for every simple decorator, and let emitter has a way to use new features from different typespec libraries without waiting tcgc support and bump tcgc version. then how to use that feature is left to emitter.

Reply:
@jhendrixMSFT the value of `Typespec.Xml.@attribute` is the fully qualified name of the decorator.  Do you have another suggestion on how to model this?

Reply:
Ok great, thanks for clarifying. Can we document this?

Reply:
@jhendrixMSFT do you mean i need to doc how each decorator in the list should be used in emitter, iow add link to that decorator doc page?

Reply:
I mean document the structure of the data within the record. i.e. the key is the fully qualified name of a decorator, and the value is the decorator's JSON representation (I assume that's what it is based on the tests). It would also be good to mention that this is constrained by an allow list which means the record will be empty if none of the allowed decorators have been applied (i.e. this doesn't contain all decorators).

Reply:
Added. Could you help to review again?

Reply:
Thanks!